src/app/sign-in.component.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+
+@Component({selector: 'sign-in', templateUrl: './sign-in.component.html'})
+export class SignInComponent implements OnInit {
+  constructor(
+      private router: Router
+  ) {}
+
+  isLoaded: boolean = false;
+  isSignedIn: boolean = false;
+
+  signIn(e: Event): void {
+    gapi.auth2.getAuthInstance().signIn();
+  }
+
+  signOut(e: Event): void {
+    gapi.auth2.getAuthInstance().signOut();
+  }
+
+  ngOnInit(): void {
+    gapi.load('auth2', this.handleAuth2Loaded.bind(this))
+  }
+
+  handleAuth2Loaded(): void {
+    gapi.auth2.init({
+      client_id: '887440561153-pb9gmue2cbbs2gbn9nkr35g0ifpvb8g5.apps.googleusercontent.com',
+      hosted_domain: 'pmi-ops.org'
+    }).then(this.handleAuth2Initialized.bind(this), this.handleAuth2Error.bind(this))
+  }
+
+  handleAuth2Initialized(): void {
+    this.isLoaded = true;
```

Comment:
I think the issue here is that the gapi async callback breaks Angular's change detection.

Here's more detail on Angular change detection:
https://stackoverflow.com/questions/34827334/triggering-angular2-change-detection-manually
https://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html
This makes specific mention of gapi v. Angular change detection:
https://stackoverflow.com/questions/40841197/view-not-updating-after-change-in-google-api-callback

Here's a suggestion for setting up gapi (more an alternate form than directly addressing this issue): https://stackoverflow.com/a/43412733

Overall, it seems like this is an annoying instance of Angular's magic breaking. If I hadn't used Angular before I'd be rather worried that we ran into a mysterious error so soon; but since I haven't run up against stuff like this before, I'm not as concerned. In my experience, and according to docs, Angular change detection _does_ work right with usual XHRs -- gapi just disagrees with it.